tests/Unit/IO/Test_StellarCollapseEos.cpp

```diff
@@ -0,0 +1,353 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <string>
+#include <vector>
+
+#include "DataStructures/BoostMultiArray.hpp" // IWYU pragma: keep
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Header.hpp"  // IWYU pragma: keep
+#include "IO/H5/StellarCollapseEos.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+// IWYU pragma: no_include <boost/iterator/iterator_facade.hpp>
+// IWYU pragma: no_include <boost/multi_array.hpp>
+// IWYU pragma: no_include <boost/multi_array/base.hpp>
+// IWYU pragma: no_include <boost/multi_array/extent_gen.hpp>
+// IWYU pragma: no_include <boost/multi_array/multi_array_ref.hpp>
+// IWYU pragma: no_include <boost/multi_array/subarray.hpp>
+
+namespace {
+
+// For 1D arrays the numbers (in, for example, logrho_expected) are the first
+// and second numbers in the corresponding dataset. For non-zero 3D arrays with
+// no repeating numbers they are the first 2x2x2 block
+// (all elements with indices [0][0][0] -> [1][1][1])
+// For 3D arrays with every element equal to zero (or with repeating numbers),
+// they are random numbers in the range [-10.0, 10.0]
+void test_tabulated(const std::string& file_path,
+                    const std::string& subgroup_path) noexcept {
+  h5::H5File<h5::AccessType::ReadOnly> sample_file(file_path);
+  const auto& sample_data =
+      sample_file.get<h5::StellarCollapseEos>(subgroup_path);
+
+  // The group /sample_data has the additional dataset "test_data" which
+  // is read to verify that we are not in "/"
+  if (subgroup_path != "/") {
+     CHECK(sample_data.get_scalar_dataset<double>("test_data") == 317);
+  }
+
+  CHECK(sample_data.get_scalar_dataset<double>("energy_shift") == 317);
+  CHECK(sample_data.get_scalar_dataset<int>("have_rel_cs2") == 1);
+  CHECK(sample_data.get_scalar_dataset<int>("pointsrho") == 391);
+  CHECK(sample_data.get_scalar_dataset<int>("pointstemp") == 163);
+  CHECK(sample_data.get_scalar_dataset<int>("pointsye") == 66);
+
+  const std::vector<double> logrho_expected = {3.0239960056064277,
+                                               3.0573293389397609};
+  const auto logrho_from_file = sample_data.get_rank1_dataset("logrho");
+  CHECK(logrho_from_file == logrho_expected);
+
+  const std::vector<double> logtemp_expected = {-3.0, -2.9666666666666668};
+  const auto logtemp_from_file = sample_data.get_rank1_dataset("logtemp");
+  CHECK(logtemp_from_file == logtemp_expected);
+
+  const std::vector<double> ye_expected = {0.005, 0.015};
+  const auto ye_from_file = sample_data.get_rank1_dataset("ye");
+  CHECK(ye_from_file == ye_expected);
+
+  boost::multi_array<double, 3> Abar_expected(boost::extents[2][2][2]);
```

Comment:
camel_case this (and more below)

Reply:
The name of the dataset in the file is capitalized so that is why I didn't do that. (Also I think you mean snake_case)

Reply:
(oooops yes I meant snake_case)